### PR TITLE
add likes to the article

### PR DESCRIPTION
**Pull request by meikuam**
add backend code for likes (tricks)

### CODE REVIEW

File: news/models.py [hunk 1]
```diff
@@ -76,4 +77,11 @@ def __str__(self):
 
 	class Meta:
 		verbose_name = "Content manager category"
-		verbose_name_plural = "Content manager categories"
\ No newline at end of file
+		verbose_name_plural = "Content manager categories"
+
+
```
Comment (asb29) [inline news/models.py]: too many lines

File: news/templates/news/article_detail.html [hunk 2]
```diff
@@ -25,6 +25,12 @@
 					<h1><a href="#">{{ article.title }}</a></h1>
 					{{ article.content|safe }}
 				</div>
+				<p>This artickle habe {{likes}} tricks</p>
```
Comment (asb29) [inline news/templates/news/article_detail.html]: need spaces around variable, like {{ likes }}

File: news/templates/news/article_detail.html [hunk 3]
```diff
@@ -25,6 +25,12 @@
...
+				<p>{{liked}}</p>
```
Comment (asb29) [inline news/templates/news/article_detail.html]: spaces here too

File: news/templates/news/article_detail.html [hunk 4]
```diff
@@ -25,6 +25,12 @@
...
+				<form action="like" method="POST">
```
Comment (asb29) [inline news/templates/news/article_detail.html]: better change action to  django's {% url %} (check other template files for examples)

File: news/views.py [hunk 5]
```diff
@@ -30,6 +30,8 @@ def get_context_data(self, **kwargs):
 		context = super(ArticleDetailView, self).get_context_data(**kwargs)
 		context['form'] = CommentForm()
 		context['comments'] = Comment.objects.filter(news_article=self.object)
+		context['likes'] = Like.objects.filter(article=self.object).__len__()
```
Comment (asb29) [inline news/views.py]: len()

File: news/views.py [hunk 6]
```diff
@@ -30,6 +30,8 @@ def get_context_data(self, **kwargs):
...
+		context['liked'] = 'True' if Like.objects.filter(article=self.object, author=self.request.user).__len__() != 0 else 'False'
```
Comment (asb29) [inline news/views.py]: len()

File: news/views.py [hunk 7]
```diff
@@ -83,3 +85,13 @@ def reviewcomment(request, comment_id):
...
+def like(request, slug):
+	article = get_object_or_404(Article, slug=slug)
+	if request.method == 'POST':
+		like_ = Like.objects.filter(article=article, author=request.user)
+		if like_.__len__() == 0:
+			like_ = Like(article=article, author=request.user)
+			like_.save()
+		else:
+			like_.delete()
+	return redirect(article, slug)
```
Comment (asb29) [inline news/views.py]: add newline at the end of the file

File: news/views.py [hunk 8]
```diff
@@ -83,3 +85,13 @@ def reviewcomment(request, comment_id):
 	return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
 
 
+def like(request, slug):
+	article = get_object_or_404(Article, slug=slug)
+	if request.method == 'POST':
+		like_ = Like.objects.filter(article=article, author=request.user)
+		if like_.__len__() == 0:
```
Comment (asb29) [inline news/views.py]: len()

**Review by asb29 — CHANGES_REQUESTED** [review]
fix some minor stuff i pointed in comments

File: news/templates/news/article_detail.html (hunk 4 above)
Comment (asb29) [reply news/templates/news/article_detail.html]: also fix typos line 28

File: news/views.py [hunk 9]
```diff
@@ -83,3 +85,13 @@ def reviewcomment(request, comment_id):
 	return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
```
Comment (asb29) [inline news/views.py:95]: add login_required decorator, same as in comment view

File: news/models.py (hunk 1 above)
Comment (meikuam) [reply news/models.py]: lines removed.

File: news/templates/news/article_detail.html (hunk 2 above)
Comment (meikuam) [reply news/templates/news/article_detail.html]: changed in next commit.

File: news/templates/news/article_detail.html (hunk 3 above)
Comment (meikuam) [reply news/templates/news/article_detail.html]: changed this too.

**Pull request merged**